src/connectedk8s/azext_connectedk8s/_troubleshootutils.py

```diff
@@ -0,0 +1,992 @@
+    try:
+
+        # Path to add the connected_cluster resource
+        connected_cluster_resource_path = os.path.join(filepath_with_timestamp, "Connected_cluster_resource.txt")
+        if storage_space_available:
+
+            # If storage space is available then obly store the connected cluster resource
+            with open(connected_cluster_resource_path, 'w+') as cc:
+                cc.write(str(connected_cluster))
+        return consts.Passed, storage_space_available
+
+    # For handling storage or OS exception that may occur during the execution
+    except OSError as e:
+        if "[Errno 28]" in str(e):
+            storage_space_available = False
+            telemetry.set_exception(exception=e, fault_type=consts.No_Storage_Space_Available_Fault_Type, summary="No space left on device")
+            shutil.rmtree(filepath_with_timestamp, ignore_errors=False, onerror=None)
+        else:
+            logger.warning("An exception has occured while trying to store the connected cluster resource from the cluster. Exception: {}".format(str(e)) + "\n")
+            telemetry.set_exception(exception=e, fault_type=consts.Connected_Cluster_Resource_Fault_Type, summary="Error occured while fetching the Get output of connected cluster")
+            diagnoser_output.append("An exception has occured while trying to store the connected cluster resource from the cluster. Exception: {}".format(str(e)) + "\n")
+
+    # To handle any exception that may occur during the execution
+    except Exception as e:
+        logger.warning("An exception has occured while trying to store the connected cluster resource logs from the cluster. Exception: {}".format(str(e)) + "\n")
+        telemetry.set_exception(exception=e, fault_type=consts.Connected_Cluster_Resource_Fault_Type, summary="Eror occure while storing the connected cluster resource logs")
+        diagnoser_output.append("An exception has occured while trying to store the connected cluster resource logs from the cluster. Exception: {}".format(str(e)) + "\n")
+
+    return consts.Failed, storage_space_available
+
+
+def retrieve_arc_agents_logs(corev1_api_instance, filepath_with_timestamp, storage_space_available):
+
+    global diagnoser_output
+    try:
+
+        if storage_space_available:
+            # To retrieve all of the arc agents pods that are presesnt in the Cluster
+            arc_agents_pod_list = corev1_api_instance.list_namespaced_pod(namespace="azure-arc")
+
+            # Traversing thorugh all agents
+            for each_agent_pod in arc_agents_pod_list.items:
+
+                # Fethcing the current Pod name and creating a folder with that name inside the timestamp folder
+                agent_name = each_agent_pod.metadata.name
+                arc_agent_logs_path = os.path.join(filepath_with_timestamp, "Arc_Agents_logs")
+                try:
+                    os.mkdir(arc_agent_logs_path)
+                except FileExistsError:
+                    pass
```

Comment:
In this case for specific arc agents inside timestamp folder, if fileexists we should delete it and create fresh right. I agree it's a very corner case which is not technically possible but if we don't delete it we will be appending logs to older existing one correct?